### PR TITLE
move out Test from the sysimage

### PR DESCRIPTION
**Pull request by KristofferC**
There doesn't seem to be any noticeable latency from this. Using a test file:

```julia
using Test
@testset "test" begin
    @test 1==1
end
```

With this PR I get:

```
❯ time ./julia --startup-file=no time_test.jl  
Test Summary: | Pass  Total  Time
test          |    1      1  0.0s
./julia --startup-file=no time_test.jl  0.42s user 0.77s system 447% cpu 0.266 total
```

and on 1.9:

```
❯ time julia --startup-file=no time_test.jl 
Test Summary: | Pass  Total  Time
test          |    1      1  0.0s
julia --startup-file=no time_test.jl  0.40s user 0.75s system 511% cpu 0.225 total
```

### CODE REVIEW

File: stdlib/Test/src/precompile.jl [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+precompile(Tuple{typeof(Base.CoreLogging.shouldlog), Test.TestLogger, Base.CoreLogging.LogLevel, Module, Symbol, Symbol})
+precompile(Tuple{typeof(Base.CoreLogging.handle_message), Test.TestLogger, Base.CoreLogging.LogLevel, String, Module, Symbol, Symbol, String, Int})
+precompile(Tuple{typeof(Test.detect_ambiguities), Any})
+precompile(Tuple{typeof(Test.collect_test_logs), Function})
+precompile(Tuple{typeof(Test.do_broken_test), Test.ExecutionResult, Any})
+precompile(Tuple{typeof(Test.record), Test.DefaultTestSet, Union{Test.Error, Test.Fail}})
+precompile(Tuple{typeof(Test.filter_errors), Test.DefaultTestSet})
+# TODO: Check which of the explicit precompile statments above are still needed
+# with the workload below active.
+@testset "example" begin
```
Comment (KristofferC) [inline stdlib/Test/src/precompile.jl:29]: This prints to stderr which is not so pretty

**Comment by KristofferC** [comment]
I wonder how the precompile tests that fail are relying on that Test is in the sysimage.

File: stdlib/Test/src/precompile.jl (hunk 1 above)
Comment (IanButterworth) [reply stdlib/Test/src/precompile.jl:29]: Can stderr be redirected or would that ruin the precompilation effectiveness?

Comment (KristofferC) [reply stdlib/Test/src/precompile.jl:29]: They all print from the same worker so maybe there is some test running a bunch of processes with `--compiled-modules=no` that loads Test?

Comment (KristofferC) [reply stdlib/Test/src/precompile.jl:29]: It is also printed twice when compiling Julia:

```
    JULIA stdlib/DelimitedFiles.release.image
    JULIA stdlib/LibSSH2_jll.release.image
    JULIA stdlib/SharedArrays.release.image
    JULIA stdlib/Profile.release.image
Test Summary: | Pass  Total  Time
example       |    1      1  0.1s
    JULIA stdlib/Test.release.image
    JULIA stdlib/SparseArrays.release.image
Test Summary: | Pass  Total  Time
example       |    1      1  0.1s
```

Seems it should only be once (when it is compiled)?

Comment (KristofferC) [reply stdlib/Test/src/precompile.jl:29]: Oh yeah, one with bounds and one without. So two total

Comment (KristofferC) [reply stdlib/Test/src/precompile.jl:29]: (it's the loading tests that prints it)

Comment (IanButterworth) [reply stdlib/Test/src/precompile.jl]: Perhaps all these precompiles should be guarded to happen during precompilation only like SnoopCompile ones are?

**Pull request merged**